packages/ketcher-react/src/script/ui/action/index.js

```diff
@@ -222,6 +222,20 @@ const config = {
     },
     hidden: (options) => isHidden(options, 'any-atom')
   },
+  // Removing from what should be saved - structure, which was added to paste tool,
+  // but not yet rendered on canvas
+  'remove-not-rendered-struct': {
+    action: {
+      thunk: (dispatch) => {
+        const savedSelectedTool = SettingsManager.selectionTool
```

Comment:
I propose to move this into separate function as this logic is already used in `src/script/ui/state/hotkeys.ts` in `removeNotRenderedStruct`.
May be it is worth to remove this logic from before mentioned file and put it here.